src/opnsense/scripts/unbound/blocklists.py

```diff
@@ -141,12 +141,36 @@ def uri_reader(uri):
                     'blocklist download %(uri)s (lines: %(lines)d exclude: %(skip)d block: %(blocklist)d)' % file_stats
                 )
 
+        # fetch configured DNS response type
+        localdata = None
+        localzone = None
+        if cnf.has_section('dnsbl'):
+            if cnf.has_option('dnsbl', 'localdata'):
+                localdata = bool(cnf.getint('dnsbl', 'localdata'))
+            if cnf.has_option('dnsbl', 'localzone'):
+                localzone = cnf.get('dnsbl', 'localzone').strip()
+            # localdata not supported with always_* zone types
+            if localdata and localzone.startswith('always'):
+                localdata = 0
```

Comment:
This manipulates model data. Not great for maintenance or extensibility. Would be better to add constraints to model itself.

Reply:
I admit that I was not really happy with this since it results in a UI checkbox that does nothing for some settings but didn't know how to address that in the UI.   I'm more than happy to move this into the model.  I'll look for a similar example in the code for inspiration but would be appreciative if you could point me to a best-practice example that you'd like me to follow.

In the UI I've currently set the localzone option as a dropdown.   Do you think it would be better to expose this as radio buttons since the list is short?

Reply:
I was looking into the model documentation and examples and I'm not finding any standard constraints that appear to support the required logic.  I need to either ensure that the localdata setting is unset (set to false) if the localzone setting is default, always_null or always_nxdomain, or alternatively the localdata setting should only be enabled if the localzone setting is redirect, inform_redirect or inform, and otherwise set to false.

The SetIfConstraint logic seems mainly intended to ensure a field is set if another field has a given value but there doesn't seem to be a way to specify what value the field should be set to.  I could reverse the logic and use the boolean to disable adding custom data, but I would rather not as I was trying to set this up to be extended with the ability to add user-specified responses later on.

At this point I suspect the logic in blocklist.py might be the easiest way to implement this since adding records is only supported for certain zone types.

@fichtner How would you suggest I proceed?

Reply:
@shonjir @fichtner may `DependConstraint` work if we change logic a bit: replace "DNSBL Null Response" checkbox with "Block subdomains" checkbox (or some). and then use this field to constrain the selection of Response (zones) Types?

Reply:
Whether we create two constrained lists or roll it into a single list the net effect is the same - we'd need to embed the boolean flag into the option values (think "inform:null" vs "inform:nxdomain") and add logic to blocklist.py to take action based on the embedded flag value.   The net effect is that we're really just passing the boolean value in a different way and moving the test to a slightly different place, but it does avoid passing invalid combinations to blocklist.py that need to be tested for there. 

It's pretty crufty in my opinion to embed flags in what are otherwise clean option values since you have to make some assumptions as to option format that you didn't need to before.  It would be cleaner if the necessary conditions and/or target values could be specified directly in the constraints.

Reply:
agree. since it still need to hide the localdata checkbox at certain localzone option values, maybe it makes sense to add some js-magic to the .volt to set localdata<->localzone dependencies? then the data will get into the model already "clean"

Reply:
and we can try to extend model with `AllowIfConstraint`. @fichtner are you talking about it?)

Reply:
something like https://github.com/kulikov-a/core/commit/0fb2108f1694bfa445a9a6b5ad127c81abf9b422 ?